toolkit/tools/imagecustomizer/docs/configuration.md

```diff
@@ -0,0 +1,539 @@
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+```
+
+### Services [[Services](#services)]
+
+Options for configuring systemd services.
+
+```yaml
+SystemConfig:
+  Services:
+    Enable:
+    - sshd
+```
+
+## User
+
+Options for configuring a user account.
+
+### Name [string]
+
+Required.
+
+The name of the user.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+```
+
+### UID [int]
+
+The ID to use for the user.
+This value is not used if the user already exists.
+
+Valid range: 0-60000
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    UID: 1000
+```
+
+### PasswordHashed [bool]
+
+Default: `false`.
+
+When set to true, specifies that the password provided by either `Password` or
+`PasswordPath` has already been hashed and may be copied directly into the
+`/etc/shadow` file.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    # Generated by:
+    #   PASSWORD="password"
+    #   SALT=$(tr -dc "A-Za-z0-9" < /dev/urandom 2> /dev/null | head -c 12)
+    #   openssl passwd -6 -salt "$SALT" "$PASSWORD"
+    Password: "$6$XH9YwqAMPohT$YQ0fqon.KOXz9AfjP5LE6VHifnNcsIgxmeX/iM5VF1GpFJTOpnTY.UGVRA.Xb8gYdVFqkYnnpJwlaIU1LhNHB/"
+    PasswordHashed: true
+```
+
+Note: Modern GPUs have gotten incredibly good at brute forcing hashed passwords.
+While hashing passwords is still considered best practice, unless the password is
+incredibly strong (32+ randomly generated characters), then it is recommended
+that you treat a hashed password with the same care as a plain-text password.
+
+### Password [string]
+
+Sets the user's password.
+
+Use of this property is strongly discouraged, except when debugging.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    Password: testpassword
+```
+
+### PasswordPath [string]
+
+Sets the user's password.
+The password is read from the file path specified.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    PasswordPath: test-password.txt
+```
+
+### PasswordExpiresDays [int]
+
+The number of days until the password expires and the user can no longer login.
+
+Valid range: 0-99999. Set to -1 to remove expiry.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    PasswordPath: test-password.txt
+    PasswordHashed: true
+    PasswordExpiresDays: 120
+```
+
+### SSHPubKeyPaths [string[]]
+
+File paths to SSH public key files.
+These public keys will be copied into the user's `~/.ssh/authorized_keys` file.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    SSHPubKeyPaths:
+    - id_ed25519.pub
+```
+
+### PrimaryGroup [string]
+
+The primary group of the user.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    PrimaryGroup: testgroup
+```
+
+### SecondaryGroups [string[]]
+
+Additional groups to assign to the user.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+    SecondaryGroups:
+    - sudo
+```
+
+### StartupCommand [string]
```

Comment:
Is this implemented yet?

Reply:
Yes. The implementation was taken from the imager tool.